File: frontend/src/api/places.ts
import api from "@/lib/api";

export interface PlaceSuggestion {
  place_id: string;
  label: string;
  description: string;
}

export interface ResolvedPlace {
  place_id: string;
  birth_place: string;
  display_name: string;
  latitude: number;
  longitude: number;
  timezone: string;
  country?: string | null;
  state?: string | null;
}

export const placesApi = {
  autocomplete: async (query: string, limit = 5): Promise<PlaceSuggestion[]> => {
    const response = await api.get<{ items: PlaceSuggestion[] }>("/places/autocomplete", {
      params: { q: query, limit },
    });
    return response.data.items;
  },

  resolve: async (payload: { place_id?: string; query?: string }): Promise<ResolvedPlace> => {
    const response = await api.post<ResolvedPlace>("/places/resolve", payload);
    return response.data;
  },
};

File: backend/app/schemas/places.py
"""Place resolution API schemas."""

from __future__ import annotations

from pydantic import Field

from backend.app.schemas.common import BaseSchema


class PlaceSuggestionResponse(BaseSchema):
    place_id: str
    label: str
    description: str


class PlaceAutocompleteResponse(BaseSchema):
    items: list[PlaceSuggestionResponse]


class PlaceResolveRequest(BaseSchema):
    place_id: str | None = Field(default=None, min_length=1, max_length=64)
    query: str | None = Field(default=None, min_length=2, max_length=512)


class PlaceResolveResponse(BaseSchema):
    place_id: str
    birth_place: str
    display_name: str
    latitude: float
    longitude: float
    timezone: str
    country: str | None = None
    state: str | None = None
